src/LibYear/Output/JsonOutput.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Text.Json;
+using LibYear.Core;
+using Spectre.Console;
+
+namespace LibYear.Output;
+
+internal sealed class JsonOutput : IOutput
+{
+	private readonly IAnsiConsole _console;
+
+	public JsonOutput(IAnsiConsole console)
+	{
+		_console = console;
+	}
+
+	public void DisplayAllResults(SolutionResult allResults, bool quietMode)
```

Comment:
I would love to see how much of the auxiliary classes could be eliminated by inlining a lot of the logic here:
-  I believe indented vs minimized is a formatter option on the base `JsonSerializer`
- keep PascalCase property names, rather than convert to camelCase
- since we're doing so much manual property mapping already (e.g. `PackageResult`), having that  here might improve legibility / reduce indirection, aka could we just populate an anonymous class here instead of having all those mapping classes doing the same thing implicitly?

Reply:
Using anonymous class just also means that you can't get away with using a source generated JSON serializer context which is why I didn't go down this path but happy to make this change!

Reply:
Other reason I selfishly did this though is it meant there was certainty in the json output data so that I could easily extract it for my own code when it came time to use this.